cmd/collector/app/server/grpc.go

```diff
@@ -52,9 +53,9 @@ func StartGRPCServer(params *GRPCServerParams) (*grpc.Server, error) {
 		MaxConnectionAgeGrace: params.MaxConnectionAgeGrace,
 	}))
 
-	if params.TLSConfig.Enabled {
+	if params.TLSConfig != nil {
```

Comment:
please see the PRs I linked before - this code now becomes much simpler, we just use OTEL helpers to create listener and server, and TLS is handled automatically